controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -598,6 +609,14 @@ func (r *BareMetalHostReconciler) actionInspecting(prov provisioner.Provisioner,
 		return recordActionFailure(info, metal3v1alpha1.InspectionError, provResult.ErrorMessage)
 	}
 
+	// Delete inspect annotation if exists
+	if hasInspectAnnotation(info.host) {
+		delete(info.host.Annotations, inspectAnnotationPrefix)
+		if err := r.Update(context.TODO(), info.host); err != nil {
+			return actionError{errors.Wrap(err, "failed to remove inspect annotation from host")}
+		}
```

Comment:
We might want to just return here, because if we return actionUpdate or actionComplete below the write will fail.

Reply:
Yep, that's what I meant.

Reply:
It is added now!